src/lib/hooks/useTimeAgo/index.ts

```diff
@@ -0,0 +1,45 @@
+import { useEffect, useState } from 'react';
+import { format, TDate } from 'timeago.js';
+
+/**
+ *  Accepts a date and returns a human-friendly string
+ *  for how long ago or how soon from now the date is
+ *
+ *  @param date     - the date to reformat
+ *  @param realtime - a bool for automatically updating the formatted date
+ *                    using a timeout
+ *  @returns        - the given date formatted in terms of how long ago or
+ *                    how soon from now it is
+ */
+export function useTimeAgo(date: TDate, realtime?: boolean): TDate {
+    const [timeAgo, setTimeAgo] = useState<TDate>(format(date));
+
+    useEffect(() => {
+        let timeout: NodeJS.Timeout;
```

Comment:
`NodeJS.Timeout` prob isn't right since this will run in the browser, I think you can just use `number` and make sure to use `window.setTimeout` instead of just `setTimeout`.